sdk/ai/azure-ai-inference/src/main/java/com/azure/ai/inference/ChatCompletionClientTracer.java

```diff
@@ -0,0 +1,638 @@
+            : configuration.get(CAPTURE_MESSAGE_CONTENT);
+        this.tracer = createTracer(clientOptions);
+    }
+
+    /**
+     * Traces the synchronous convenience API - {@link com.azure.ai.inference.ChatCompletionsClient#complete(ChatCompletionsOptions)}.
+     *
+     * @param request input options containing chat options for complete API.
+     * @param operation the operation performing the actual completion call.
+     * @param completeRequest The completeRequest parameter for the {@code operation}.
+     * @param requestOptions The requestOptions parameter for the {@code operation}.
+     * @return chat completions for the provided chat messages.
+     */
+    @SuppressWarnings("try")
+    ChatCompletions traceSyncComplete(ChatCompletionsOptions request, SyncCompleteOperation operation,
+        BinaryData completeRequest, RequestOptions requestOptions) {
+        if (!tracer.isEnabled()) {
+            return operation.invoke(completeRequest, requestOptions);
+        }
+        final Context span
+            = tracer.start(spanName(request), new StartSpanOptions(SpanKind.CLIENT), parentSpan(requestOptions));
+        if (tracer.isRecording(span)) {
+            traceCompletionRequestAttributes(request, span);
+            traceCompletionRequestEvents(request.getMessages(), span);
+        }
+
+        try (AutoCloseable ignored = tracer.makeSpanCurrent(span)) {
+            final ChatCompletions response = operation.invoke(completeRequest, requestOptions.setContext(span));
+            if (tracer.isRecording(span)) {
+                traceCompletionResponseAttributes(response, span);
+                traceCompletionResponseEvents(response, span);
+            }
+            tracer.end(null, null, span);
+            return response;
+        } catch (Exception e) {
+            tracer.end(null, e, span);
+            sneakyThrows(e);
+        }
+        return null;
+    }
+
+    /**
+     * Traces the convenience API - {@link com.azure.ai.inference.ChatCompletionsAsyncClient#complete(ChatCompletionsOptions)}.
+     *
+     * @param request input options containing chat options for complete API.
+     * @param operation the operation performing the actual completion call.
+     * @param completeRequest The completeRequest parameter for the {@code operation}.
+     * @param requestOptions The requestOptions parameter for the {@code operation}.
+     * @return chat completions for the provided chat messages.
+     */
+    Mono<ChatCompletions> traceComplete(ChatCompletionsOptions request, CompleteOperation operation,
+        BinaryData completeRequest, RequestOptions requestOptions) {
+        if (!tracer.isEnabled()) {
+            return operation.invoke(completeRequest, requestOptions);
+        }
+
+        final Mono<Context> resourceSupplier = Mono.fromSupplier(() -> {
+            final Context span
+                = tracer.start(spanName(request), new StartSpanOptions(SpanKind.CLIENT), parentSpan(requestOptions));
```

Comment:
super-nit: we can probably optimize a bit by putting `new StartSpanOptions(SpanKind.CLIENT)` into a constant - it's de-facto immutable

Reply:
Yep, moved to static final now